graph/graph.go

```diff
@@ -112,11 +113,18 @@ func NewGraph(root string, driver graphdriver.Driver) (*Graph, error) {
 	}
 
 	graph := &Graph{
-		root:     abspath,
-		idIndex:  truncindex.NewTruncIndex([]string{}),
-		driver:   driver,
-		retained: &retainedLayers{layerHolders: make(map[string]map[string]struct{})},
+		root:             abspath,
+		idIndex:          truncindex.NewTruncIndex([]string{}),
+		driver:           driver,
+		retained:         &retainedLayers{layerHolders: make(map[string]map[string]struct{})},
+		tarSplitDisabled: false,
 	}
+
+	// Windows does not currently support tarsplit functionality.
+	if runtime.GOOS == "windows" {
+		graph.tarSplitDisabled = true
```

Comment:
Tar-split does do the assembly of an archive from access to a relative path.

Is this windows registry hive delta basically a block level delta?
On Aug 7, 2015 16:41, "John Starks" <notifications@github.com> wrote:

> In graph/graph.go
> <https://github.com/stevvooe/docker/pull/1#discussion_r36559225>:
>
> >  	}
> > +
> > +	// Windows does not currently support tarsplit functionality.
> > +	if runtime.GOOS == "windows" {
> > +		graph.tarSplitDisabled = true
>
> A bigger part of the problem is that our transport format needs to include
> the Windows registry hive delta files, and those are not available on the
> host via the path returned from Get(). If I understand the tar-split code
> correctly (and I probably don't :)), it relies on being able to get all the
> file contents via direct file system access from Get().
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevvooe/docker/pull/1/files#r36559225>.
>